cpp/src/io/parquet/compact_protocol_writer.cpp

```diff
@@ -201,7 +201,20 @@ size_t CompactProtocolWriter::write(const ColumnChunkMetaData& s)
   c.field_int(9, s.data_page_offset);
   if (s.index_page_offset != 0) { c.field_int(10, s.index_page_offset); }
   if (s.dictionary_page_offset != 0) { c.field_int(11, s.dictionary_page_offset); }
-  if (s.statistics_blob.size() != 0) { c.field_struct_blob(12, s.statistics_blob); }
+  // if (s.statistics_blob.size() != 0) { c.field_struct_blob(12, s.statistics_blob); }
+  c.field_struct(12, s.statistics_blob);
+  return c.value();
+}
+
+size_t CompactProtocolWriter::write(const Statistics& s)
+{
+  CompactProtocolFieldWriter c(*this);
+  if (s.max.size() != 0) { c.field_binary(1, s.max); }
+  if (s.min.size() != 0) { c.field_binary(2, s.min); }
+  if (s.null_count != -1) { c.field_int(3, s.null_count); }
+  if (s.distinct_count != -1) { c.field_int(4, s.distinct_count); }
+  if (s.max_value.size() != 0) { c.field_binary(5, s.max_value); }
+  if (s.min_value.size() != 0) { c.field_binary(6, s.min_value); }
```

Comment:
I'm surprised to see a writer change here - was there something incorrect about the way we've been writing row group statistics?

Reply:
No. The statistics are pre-encoded during page encoding itself and stored as binary blob. I changed the datatype from binary blob to `Statistics` struct. So, this change is encoding that struct again.